primefaces/src/main/java/org/primefaces/component/datatable/feature/FilterFeature.java

```diff
@@ -195,9 +195,11 @@ public void filter(FacesContext context, DataTable table) {
                 FilterConstraint constraint = filter.getConstraint();
                 Object filterValue = filter.getFilterValue();
                 if (filterValue instanceof String && column instanceof ColumnBase) {
-                    ColumnBase columnBase = (ColumnBase) column;
-                    if (columnBase.getConverter() != null) {
-                        filterValue = columnBase.getConverterObject().getAsObject(context, columnBase, (String) filterValue);
+                    try {
+                        filterValue = ComponentUtils.getConvertedValue(context, (ColumnBase) column, filterValue);
```

Comment:
instead of accessing ColumnBase in the very generic ComponentUtils, maybe pass the converter via param and overload the method? Or even lambda if we need it lazy